### DISCUSSION TITLE
Just one bin.install requires full Command Line Tools (CLT) for Xcode

### DISCUSSION THREAD

**Discussion by aleksandr-vin** [Homebrew/discussions#3211] [Everyday usage]
Hello, I've made a [simple formula](https://github.com/aleksandr-vin/homebrew-taps/blob/main/Formula/klm-dictionary.rb) to install one shell script into bin. And it requires a user to install full CLT.

Why it is needed to handle copying of one file only?

**Comment by SMillerDev** [comment]
Because the CLT is required to use Homebrew.

**Reply by Bo98** [reply]
If you ship a bottle, you can avoid the build tools dependency (except Apple Silicon - but that'll likely change soon).

When building from source, Homebrew requires build tools. Homebrew does not do static analysis of build scripts.

**Reply by aleksandr-vin** [reply]
Thanks! So I need to pack same script into bottles for all the supported archs?

**Reply by gromgit** [reply]
If I remember correctly, `brew bottle` is smart enough to detect when all the binaries it's bottling are scripts or other non-arch-specific artifacts, and create a single `all`-arch bottle, so you don't have to run it N times on N different OSes.
